### DISCUSSION TITLE
problem with format "msh4" and quad dominant mesh

### DISCUSSION THREAD

**Discussion by damulaoshi** [neperfepx/neper#595] [Q&A] [answered]
Dear Romain,
I have a problem whenever I want to generate the .msh4 format mesh file, the code:
neper -T -n 200 -dim 3 -domain "cube(50,50,50)" -o mesh_test;
neper -M mesh_test.tess -elttype hex -o mesh_test; -> it works fine and I got:
![image](https://user-images.githubusercontent.com/75380898/198080551-1b41a21f-28d6-40aa-8be8-3e452405681b.png)
but whenever I want to generate the .msh4 file, it'll abort:
neper -M mesh_test.tess -elttype hex -format msh4 -o mesh_test; and I got the error:
![image](https://user-images.githubusercontent.com/75380898/198080898-5a3b88f7-1423-4aac-bee8-73e6dbb6dedf.png)
Why does it happen?

Another question is:
Is there any way to generate quad-dominant mesh, not full-quad mesh (for 2D)? After I saw the discussion "Quad dominant or all Quad mesh in 2D (preserving grain boundaries) #98", I've tried but I can only get full-quad mesh,but what I want is a quad-dominant mesh, is there any possibility to do that in Neper? Or is there any way that I can generate a msh4 file and then use gmsh to do the quad dominant mesh just like you said for the full-quad mesh in discussion #98?
Thanks before!

Best regards

**Comment by rquey** [comment, marked as answer]
`msh4` is not supported for hex elements.  The `msh4` format embeds topological relationships that are troublesome for the hex meshes that Neper generates...  I've added a clean error message for this.

I never went beyond #98 for what is conformal quad meshing, so I cannot answer your second question.
